flexmeasures/data/tests/test_scheduling_repeated_jobs.py

```diff
@@ -0,0 +1,206 @@
+
+
+@pytest.mark.parametrize(
+    "args_modified,kwargs_modified,equal",
+    [
+        (
+            [1, 2, "1"],
+            {
+                "key1": "value1",
+                "key2": "value2",
+                "key3": 3,
+                "key4": {"key1_nested": 1, "key2_nested": 2},
+            },
+            True,
+        ),
+        (
+            [1, 2, 1],
+            {
+                "key1": "value1",
+                "key2": "value2",
+                "key3": 3,
+                "key4": {"key1_nested": 1, "key2_nested": 2},
+            },
+            False,
+        ),
+        (
+            [1, 2, "1"],
+            {
+                "key1": "different",
+                "key2": "value2",
+                "key3": 3,
+                "key4": {"key1_nested": 1, "key2_nested": 2},
+            },
+            False,
+        ),
+        (
+            [1, 2, "1"],
+            {
+                "different": "value1",
+                "key2": "value2",
+                "key3": 3,
+                "key4": {"key1_nested": 1, "key2_nested": 2},
+            },
+            False,
+        ),
+        (
+            [1, 2, "1"],
+            {
+                "key1": "value1",
+                "key2": "value2",
+                "key3": 3,
+                "key4": {"key1_nested": "different", "key2_nested": 2},
+            },
+            False,
+        ),
+    ],
+)
+def test_hashing_simple(args_modified: list, kwargs_modified: dict, equal: bool):
+    args = [1, 2, "1"]
+    kwargs = {
+        "key1": "value1",
+        "key2": "value2",
+        "key3": 3,
+        "key4": {"key1_nested": 1, "key2_nested": 2},
+    }
+
+    hash_original = hash_function_arguments(args, kwargs)
+    hash_modified = hash_function_arguments(args_modified, kwargs_modified)
+
+    if equal:
+        assert hash_original == hash_modified
+    else:
+        assert hash_original != hash_modified
+
+
+def test_hashing(db, app, add_charging_station_assets, setup_test_data):
+    soc_at_start = 1
+    target_soc = 5
+    duration_until_target = timedelta(hours=2)
+
+    charging_station = Sensor.query.filter(
+        Sensor.name == "Test charging station"
+    ).one_or_none()
+    tz = pytz.timezone("Europe/Amsterdam")
+    start = tz.localize(datetime(2015, 1, 2))
+    end = tz.localize(datetime(2015, 1, 3))
+    target_datetime = start + duration_until_target
+    resolution = timedelta(minutes=15)
+    soc_targets = [dict(datetime=target_datetime.isoformat(), value=target_soc)]
+
+    kwargs = dict(
+        sensor=charging_station,
+        start=start,
+        end=end,
+        belief_time=start,
+        resolution=resolution,
+        flex_model={"soc-at-start": soc_at_start, "soc-targets": soc_targets},
+    )
+    args = []
+
+    hash = hash_function_arguments(args, kwargs)
+    print(hash)
+    # checks that hashes are consistent between calls
+    assert hash == "+aVb5DY1c64pu7wHl8XHvmbClu6Y9fqww8QDOrlrtCM="
```

Comment:
The assert statement doesn't match the comment. I'd keep the comment, and, instead of line 116, assert that the results of two calls to `hash_function_arguments()` (with deep copies of the args and kwargs) match.

Reply:
oh yes, well here the comment is off. This actually checks if the hash changes between different runtimes. I've also added a test to check within the same runtime.